File: 06-rest-demo-final/src/main/java/demos/springdata/restdemo/dao/UserRepository.java
package demos.springdata.restdemo.dao;

import demos.springdata.restdemo.model.Post;
import demos.springdata.restdemo.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByUsername(String username);
}

File: fmi-2023-04-ts-lab/src/dao/user-repository.ts
import { Optional } from "../common.js";
import { User } from "../model/users.js";
import { Repository, RepositoryImpl } from "./repository.js";

export interface UserRepository extends Repository<User> {
    findByUsername(username: string): Optional<User>;
}

export class UserRepositoryImpl extends RepositoryImpl<User> implements UserRepository {
    findByUsername(username: string): Optional<User> {
        return this.findAll().find(u => u.username === username);
    }
}